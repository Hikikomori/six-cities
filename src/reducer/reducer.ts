import * as Redux from 'redux'
import {reducer as base} from './base/base'
import {reducer as favorites} from './favorites/favorites'
import {reducer as reviews} from './reviews/reviews'
import {reducer as user} from './user/user'

export default Redux.combineReducers({
  BASE: base,
  FAVORITES: favorites,
  REVIEIWS: reviews,
  USER: user
})
