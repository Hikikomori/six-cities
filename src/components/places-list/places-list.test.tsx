import * as renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../../reducer/reducer';
import card from '../../mocks/card'
import PlacesList from './places-list';

const store = createStore(reducer);

it(`Main places list correctly renders`, () => {
  const tree = renderer
    .create(<Provider store={store}>
      <BrowserRouter>
        <PlacesList
          cards={[card]}
          isOnPropertyPage={false}
          isOnFavoritesPage={false}
        />
      </BrowserRouter>
    </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Property nearby places list correctly renders`, () => {
  const tree = renderer
    .create(<Provider store={store}>
      <PlacesList
        cards={[card]}
        isOnPropertyPage={true}
        isOnFavoritesPage={false}
      />
    </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Favorite places list correctly renders`, () => {
  const tree = renderer
    .create(<Provider store={store}>
      <BrowserRouter>
        <PlacesList
          cards={[card]}
          isOnPropertyPage={false}
          isOnFavoritesPage={true}
        />
      </BrowserRouter>
    </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
