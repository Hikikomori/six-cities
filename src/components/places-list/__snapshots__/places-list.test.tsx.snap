// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite places list correctly renders 1`] = `
<div
  className="favorites__places"
>
  <article
    className="favorites__card place-card"
  >
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://es31-server.appspot.com/six-cities/static/hotel/17.jpg"
        width="260"
      />
    </div>
    <div
      className="favorites__card-info place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            161
             
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Canal View Prinsengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        House
      </p>
    </div>
  </article>
</div>
`;

exports[`Main places list correctly renders 1`] = `
<div
  className="cities__places-list tabs__content places__list"
>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://es31-server.appspot.com/six-cities/static/hotel/17.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            161
             
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Canal View Prinsengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        House
      </p>
    </div>
  </article>
</div>
`;

exports[`Property nearby places list correctly renders 1`] = `
<div
  className="near-places__list places__list"
>
  <article
    className="near-places__card place-card"
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://es31-server.appspot.com/six-cities/static/hotel/17.jpg"
        width="260"
      />
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            161
             
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        Canal View Prinsengracht
      </h2>
      <p
        className="place-card__type"
      >
        House
      </p>
    </div>
  </article>
</div>
`;
